DataDog/libdatadog
# [crashtracker] Use blazesym to resolve stack trace names in the receiver

# What does this PR do?

Uses blazesym to resolve symbol addresses into names.  

# Motivation

Crash reports with names are better than crashreports with pointers.

# Additional Notes

This depends on an unreleased change in Blazesym.  We pin to a commit that has the change.  Once they release it, we can move to use that version. 

# How to test the change?

run `crashtracker::api::test_crash` and see that the the emitted crash report now includes 

after:
```
  "stacktrace": [
    {
      "ip": "0xaaaab1b32350",
      "module_base_address": null,
      "names": [
        {
          "colno": 5,
          "filename": "mod.rs",
          "lineno": 66,
          "name": "backtrace::backtrace::trace_unsynchronized"
        }
      ],
      "sp": "0xffffa5f758c0",
      "symbol_address": "0xaaaab1b3231c"
    },
    {
      "ip": "0xaaaab1b32438",
      "module_base_address": null,
      "names": [
        {
          "colno": 5,
          "filename": "collectors.rs",
          "lineno": 30,
          "name": "datadog_crashtracker::collectors::emit_backtrace_by_frames"
        }
      ],
      "sp": "0xffffa5f75900",
      "symbol_address": "0xaaaab1b32380"
    },
```

Before:
```
  "stacktrace": [
    {
      "ip": "0xaaaadd44234c",
      "module_base_address": null,
      "names": null,
      "sp": "0xffffa4b508c0",
      "symbol_address": "0xaaaadd442318"
    },
    {
      "ip": "0xaaaadd442434",
      "module_base_address": null,
      "names": null,
      "sp": "0xffffa4b50900",
      "symbol_address": "0xaaaadd44237c"
    },
```


## For Reviewers
